pkg/cmd/taskrun/delete.go

```diff
@@ -176,6 +179,25 @@ func deleteTaskRuns(s *cli.Stream, p cli.Params, trNames []string, opts *options
 		d = deleter.New("TaskRun", func(taskRunName string) error {
 			return actions.Delete(trGroupResource, cs.Dynamic, cs.Tekton.Discovery(), taskRunName, p.Namespace(), metav1.DeleteOptions{})
 		})
+		for _, trNane := range trNames {
+			tr, err := taskrun.Get(cs, trNane, metav1.GetOptions{}, p.Namespace())
+			if err != nil {
+				return fmt.Errorf("failed to get taskrun")
+			}
+
+			if !ownerPrFinished(cs, *tr) {
+				if opts.ForceDelete {
+					fmt.Fprintln(s.Out, "warning: related pipelinerun still running.")
+					break
+				}
+				fmt.Fprintf(s.Out, "TaskRun(s) attached to PipelineRun is still running deleting will restart the completed taskrun. Proceed (y/n): ")
+				err := opts.TakeInput(s, "")
```

Comment:
how we wil handle this in pruner, as we are using this as a script ?

Reply:
Pruner uses tkn with keep or keepsince which does not need a name of taskrun. and with keep,keepsince,deleteallns we by default ignore the taskruns having pipelinruns running.